src/components/structures/MatrixChat.js

```diff
@@ -361,6 +365,7 @@ module.exports = React.createClass({
                         room_id: result.room_id,
                         event_id: payload.event_id,
                         invite_sign_url: payload.invite_sign_url,
+                        room_oob_data: payload.room_oob_data,
```

Comment:
This looks inconsistent. s/room_//g ?

This highlights a general problem here - these payloads, and the arguments to `_viewRoom` are getting a bit silly. I feel like they should be factored out to a separate object which documents its fields.